packages/react-components/src/components/LocalVideoTile.tsx

```diff
@@ -54,17 +54,79 @@ export const LocalVideoTile = React.memo(
+
+    // When localVideoViewOptions change two things may happen:
+    //
+    // 1. Just the scaling mode has changed and we have access to the original LocalStreamRenderView
+    //   - In this case we just need to call updateScalingMode on the view and do not need to recreate the stream view
+    // 2. isMirrored has changed, or scalingMode change and we don't have access to the original LocalStreamRenderView
+    //   - In this case we need to dispose the old view and create a new one
+    //
+    // For scenario 1, we hold onto a ref to the scalingMode that persists across renders. When localVideoViewOptions.scalingMode
+    // differs from the persistent ref, we know to call updateScalingMode. We then subsequently update the ref to the new value for
+    // future renders.
+    //
+    // For scenario 2, we extract the localVideoViewOptions.isMirrored to ensure any change to the isMirrorred prop triggers the useEffect
+    // that recreates the local stream view. We must also here add an extra check for when the scaling mode changes and we do not have
+    // access to the original LocalStreamView (and hence cannot call updateScalingMode). When this happens we must also trigger the useEffect
+    // that recreates the local stream view.
+
+    const scalingModeRef = useRef(localVideoViewOptions?.scalingMode);
+    const newIsMirrored = localVideoViewOptions?.isMirrored;
+    const newScalingMode = localVideoViewOptions?.scalingMode;
+    const hasScalingModeChanged = scalingModeRef.current !== newScalingMode;
+    const updatingScalingModeDirectly = hasScalingModeChanged && !!localStreamRendererResult;
+
+    if (isAvailable && renderElement && newScalingMode && updatingScalingModeDirectly) {
+      localStreamRendererResult && localStreamRendererResult.view.updateScalingMode(newScalingMode);
+      scalingModeRef.current = localVideoViewOptions?.scalingMode;
+    }
+
+    // scalingModeForUseEffect will trigger the useEffect to recreate the local stream view only if the scaling mode has changed and
+    // we cannot call updateScalingMode on the view object directly. Otherwise it will be null to ensure the useEffect does not trigger
+    // when scaling mode changes.
+    const scalingModeForUseEffect = updatingScalingModeDirectly ? null : newScalingMode;
+
     useEffect(() => {
+      // Avoid race condition where onDisposeLocalStreamView is called before onCreateLocalStreamView
+      // and setLocalStreamRendererResult have completed
+      let wasLocalStreamDisposed = false;
+
       if (isAvailable && !renderElement) {
-        onCreateLocalStreamView?.(localVideoViewOptions);
+        (async (): Promise<void> => {
+          console.log('calling onCreateLocalStreamView');
+          const streamRendererResult = await onCreateLocalStreamView?.({
+            isMirrored: newIsMirrored,
+            scalingMode: scalingModeForUseEffect === null ? scalingModeRef.current : scalingModeForUseEffect
```

Comment:
Oh cool (?), so eslint allows you to use `scalingModeRef` in the `useEffect` without adding it to dependency array?

That _is_ what you want here, but I expected you to need to add an eslint exception for it.

Reply:
My guess is react has some internal knowledge that a React.MutableRef shouldn't be included in the useEffect array as it should be constant across all renders